src/providers/zerion.rs

```
@@ -271,7 +272,19 @@ impl HistoryProvider for ZerionProvider {
                     chain: if f.relationships.chain.data.r#type != "chains" {
                         None
                     } else {
-                        Some(f.relationships.chain.data.id)
+                        // Try to convert chain id from human readable to caip2 format
+                        match crypto::string_chain_id_to_caip2_format(
+                            &f.relationships.chain.data.id,
+                        ) {
+                            Ok(chain) => Some(chain),
+                            Err(_) => {
+                                info!(
+                                    "Error on parsing chain id to caip2 format: {:?}",
+                                    f.relationships.chain.data.id
+                                );
+                                Some(f.relationships.chain.data.id)
```

I don't think we should fallback to an invalid value. Instead, the entry should be excluded from the results list.

---

I don't mind changing it to excluded in this case, but passing it through, we could catch the unsupported chain id at both the backend and frontend side, assuming that on the frontend side, there is a check for the CAIP-2 format, and if it's not, just behave like it's excluded.

---

Why error in the client at all? We don't support the chain. We don't need to cause more errors

---

I'm not speaking about the error on the frontend side:

> assuming that on the frontend side, there is a check for the CAIP-2 format, and if it's not, just behave like it's excluded.

If there is a strong opinion not passing it through, I'll just exclude it in this case to unblock this feature asap.

---

[Just changed it](https://github.com/WalletConnect/blockchain-api/pull/494/commits/155db8ba40d19fa219a171512c29153f5847b8e7) to not pass through in case if it's not caip-2.